lib/puppet/configurer/fact_handler.rb

```diff
@@ -32,6 +32,6 @@ def facts_for_uploading
 
     text = facts.render(:pson)
 
-    {:facts_format => :pson, :facts => CGI.escape(text)}
+    {:facts_format => :pson, :facts => Puppet::Util.uri_query_escape(text)}
```

Comment:
just fyi this code has changed in master so this might need some fixup on merge-up